mobius-extras/src/main/java/com/spotify/mobius/extras/connections/MergeConnectablesConnection.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.extras.connections;
+
+import com.spotify.mobius.Connectable;
+import com.spotify.mobius.Connection;
+import com.spotify.mobius.functions.Consumer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MergeConnectablesConnection<A, B> implements Connection<A> {
+
+  private final List<Connection<A>> connections;
+
+  public static <A, B> Connection<A> create(
+      List<Connectable<A, B>> connectables, Consumer<B> output) {
+    return new MergeConnectablesConnection<>(connectables, output);
+  }
+
+  public static <A, B> Connection<A> create(
+      Connectable<A, B> fst, Connectable<A, B> snd, Consumer<B> output) {
+    return create(Arrays.asList(fst, snd), output);
+  }
+
+  private MergeConnectablesConnection(List<Connectable<A, B>> connectables, Consumer<B> output) {
+    connections = new ArrayList<>(connectables.size());
+    for (Connectable<A, B> connectable : connectables) {
+      connections.add(connectable.connect(output));
+    }
+  }
+
+  @Override
+  public synchronized void accept(A value) {
```

Comment:
It would be nice to not have to synchronize while calling out into user code (through `accept`). Holding on to a lock while doing that increases the risk of deadlocks. Maybe something like this might work:

1. Make `collections` not final, and a `CopyOnWriteArrayList`.
1. Make `accept` not `synchronized`.
1. Make `dispose` not `synchronized`, and make it first copy the `connections` array, then clear it, then iterate over the copied array and dispose all connections. Since `dispose` is supposed to be idempotent, it doesn't matter if two disposes race. 

The problematic case is when a dispose and an accept race with each other; it is possible to get to a state where `accept` invokes an already disposed connection. I think that's likely a better situation than getting potential deadlocks, but I'm not completely sure.